File: subprojects/docs/src/samples/scala/crossCompilation/src/test/scala/org/gradle/sample/PersonTest.scala
package org.gradle.sample

import _root_.junit.framework.TestCase
import _root_.org.gradle.sample.impl.PersonImpl
import _root_.org.gradle.sample.impl.JavaPerson

class PersonTest extends TestCase {

  def testCanCreateScalaPersonImpl(): Unit = {
    def person: Person = new PersonImpl("bob smith")
    person
  }

  def testCanCreateJavaPersonImpl(): Unit = {
    def person: Person = new JavaPerson("bob smith")
    person
  }

}

File: subprojects/docs/src/snippets/java/multiproject/kotlin/shared/src/test/java/org/gradle/shared/PersonTest.java
package org.gradle.shared;

import junit.framework.TestCase;

import java.io.IOException;
import java.util.Properties;

public class PersonTest extends TestCase {
    public void testTest() {
        Person person = new Person("testname1");
        assertEquals("testname1", person.getName());
        person.setName("testname2");
        assertEquals("testname2", person.getName());
    }

    public void testMainProperty() throws IOException {
        assertEquals("mainValue", new Person("test").readProperty());
    }

    public void testTestProperty() throws IOException {
        Properties properties = new Properties();
        properties.load(getClass().getClassLoader().getResourceAsStream("org/gradle/shared/test.properties"));
        assertEquals("testValue", properties.getProperty("test"));
    }
}
